doc/api/readline.md

```diff
@@ -7,22 +7,49 @@
 <!-- source_link=lib/readline.js -->
 
 The `readline` module provides an interface for reading data from a [Readable][]
-stream (such as [`process.stdin`][]) one line at a time. It can be accessed
-using:
+stream (such as [`process.stdin`][]) one line at a time.
 
-```js
+To use the promise-based APIs:
+
+```mjs
+import * as readline from 'readline/promises';
+```
+
+```cjs
+const readline = require('readline/promises');
+```
+
+To use the callback and sync APIs:
+
+```mjs
+import * as readline from 'readline';
+```
+
+```cjs
 const readline = require('readline');
 ```
 
 The following simple example illustrates the basic use of the `readline` module.
 
-```js
+```mjs
+import * as readline from 'readline/promises';
```

Comment:
```suggestion
import * as readline from 'node:readline/promises';
```

(I won't add suggestions for all of these... I'd just like to see us start to use the `node:` prefix consistently with ESM examples.

Reply:
Any reason to add them to ESM examples only? Shouldn't we add it to CJS examples as well?
Also, I'm not sure if https://github.com/nodejs/node/issues/38343#issuecomment-824931220 has been resolved, maybe it would be a blocking concern?